# Regressão 01 - tarefa 02 - transformações em X e Y

### 1. Melhorando o ajuste no percentual de gorjetas

a. Vamos partir do modelo feito na última tarefa, o que relaciona ```tip_pct``` e ```net_bill```. Carregue a base, os pacotes e reproduza este modelo aqui.

In [52]:
import pandas as pd
import seaborn as sns
from seaborn import load_dataset

import matplotlib.pyplot as plt

import numpy as np

import statsmodels.formula.api as smf

In [53]:
# importando a base de dados

tips = sns.load_dataset("tips")
tips['tip_pct'] = tips['tip'] / (tips['total_bill'] - tips['tip'])
tips['net_bill'] = tips['total_bill'] - tips['tip']   
tips.head()

total_bill   tip     sex smoker  day    time  size   tip_pct  net_bill
0       16.99  1.01  Female     No  Sun  Dinner     2  0.063204     15.98
1       10.34  1.66    Male     No  Sun  Dinner     3  0.191244      8.68
2       21.01  3.50    Male     No  Sun  Dinner     3  0.199886     17.51
3       23.68  3.31    Male     No  Sun  Dinner     2  0.162494     20.37
4       24.59  3.61  Female     No  Sun  Dinner     4  0.172069     20.98

In [54]:
# ajustando modelo de tip_pct em função de net_bill

reg = smf.ols('tip_pct ~ net_bill', data = tips).fit()

reg.rsquared

0.09129084920662944

### 2. ajuste o modelo de ```tip_pct``` em função do logaritmo de ```net_bill```.

In [55]:
# criando coluna com valores de lagorítmo de net_bill

tips['net_bill_log'] = np.log(tips.net_bill)

In [56]:
# ajustando modelo de tip_pct em função de logarítmo de net_bill (net_bill_log)

reg2 = smf.ols('tip_pct ~ net_bill_log', data = tips).fit()

reg2.rsquared

0.20172884237605493

###  3. ajuste o modelo de ```tip_pct``` em função de um polinômio de segundo grau em ```net_bill```.

In [57]:
# criando coluna com valores de polinômio de segundo grau de net_bill

tips['net_bill_pol'] = tips['net_bill'] + np.power(tips['net_bill'], 2)


In [58]:
# ajustando modelo de de tip_pct em função de um polinômio de segundo grau de net_bill

reg3 = smf.ols('tip_pct ~ net_bill_pol', data = tips).fit()

reg3.rsquared

0.05358049356274852

###  4. ajuste um modelo no log do ```tip_pct``` em função de ```net_bill```. Calcule o $R^2$ também em ```tip_pct```

In [59]:
# criando coluna com valores de lagorítmo de tip_pct

tips['tip_pct_log'] = np.log(tips.tip_pct)

In [60]:
# ajustando modelo de logarítmo de tip_pct (tip_pct_log) em função net_bill

reg4 = smf.ols('tip_pct_log ~ net_bill', data = tips).fit()

reg4.rsquared

0.23534487140864324

###  5. ajuste um modelo no log do ```tip_pct``` em função do log de ```net_bill```. Calcule o $R^2$ também em ```tip_pct```.

In [61]:
# ajustando modelo de logarítmo de tip_pct (tip_pct_log) em função de logarítmo de net_bill (net_bill_log)

reg5 = smf.ols('tip_pct_log ~ net_bill_log', data = tips).fit()

reg5.rsquared

0.2934740215080778

###  6. qual modelo resultou no melhor ajuste?

*O modelo que tem apesentado o melhor ajuste foi o reg5 (modelo de logarítmo de tip_pct (tip_pct_log) em função de logarítmo de net_bill (net_bill_log), que apresenta o maior R², de 0.293*